File: backend/app/schemas/vehicle.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class VehicleOut(BaseModel):
    id: UUID
    provider: str
    provider_vehicle_id: str
    vin: str | None
    display_name: str | None
    model: str | None
    year: int | None
    timezone: str
    nominal_battery_kwh: float | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class SnapshotOut(BaseModel):
    id: int
    vehicle_id: UUID
    observed_at: datetime
    battery_level: float | None
    usable_battery_level: float | None
    battery_range_km: float | None
    est_battery_range_km: float | None
    charging_state: str | None
    charge_limit_soc: float | None
    charger_power_kw: float | None
    charger_voltage: int | None
    charger_actual_current: int | None
    charger_phases: int | None
    plugged_in: bool | None
    latitude: float | None
    longitude: float | None
    speed_kmh: float | None
    odometer_km: float | None
    vehicle_state: str | None
    inside_temp_c: float | None
    outside_temp_c: float | None
    climate_on: bool | None
    sentry_mode: bool | None
    locked: bool | None

    model_config = {"from_attributes": True}


class ProviderHealthOut(BaseModel):
    healthy: bool
    last_success_at: datetime | None
    last_error: str | None
    recent_errors: int


class TeslaAuthUrlOut(BaseModel):
    auth_url: str
    state: str

File: frontend/src/types/vehicle.ts
export interface Vehicle {
  id: string;
  provider: string;
  provider_vehicle_id: string;
  vin: string | null;
  display_name: string | null;
  model: string | null;
  year: number | null;
  timezone: string;
  nominal_battery_kwh: number | null;
  created_at: string;
  updated_at: string;
}

export interface Snapshot {
  id: number;
  vehicle_id: string;
  observed_at: string;
  battery_level: number | null;
  usable_battery_level: number | null;
  battery_range_km: number | null;
  est_battery_range_km: number | null;
  charging_state: string | null;
  charge_limit_soc: number | null;
  charger_power_kw: number | null;
  charger_voltage: number | null;
  charger_actual_current: number | null;
  charger_phases: number | null;
  plugged_in: boolean | null;
  latitude: number | null;
  longitude: number | null;
  speed_kmh: number | null;
  odometer_km: number | null;
  vehicle_state: string | null;
  inside_temp_c: number | null;
  outside_temp_c: number | null;
  climate_on: boolean | null;
  sentry_mode: boolean | null;
  locked: boolean | null;
}

export interface ProviderHealth {
  healthy: boolean;
  last_success_at: string | null;
  last_error: string | null;
  recent_errors: number;
}
